test/status_translator_test.cc

```diff
@@ -0,0 +1,100 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "lib/translate/status_translator.h"
+
+#include "absl/base/casts.h"
+#include "gtest/gtest.h"
+#include "lib/translate/common.h"
+
+// Tests
+
+namespace {
+
+TEST(TranslateGenericCommandStatus, ShouldReturnCorrectStatus) {
+  uint8_t sct = 0x0;  // kGeneric
+  uint8_t sc = 0x0;   // kSuccess;
+  translator::TranslatedScsiStatus *translated =
+      new translator::TranslatedScsiStatus();
```

Comment:
It's usually a bad idea in modern C++ to ever call `new` explicitly, as it makes it easy to inadvertently create a memory leak, as is the case here. Instead you can use something like `unique_ptr` which takes care of cleanup automatically.

However, in this particular case we don't need a heap allocation at all; you can define a `TranslatedScsiStatus` on the stack and pass in it address. This is nearly always preferred when feasible, since it's faster and less error-prone.

Reply:
good general tip about the `unique_ptr`! and you're definitely right that here a stack allocated struct is sufficient